### PR TITLE
feat(admin): remove unnecessary login dialog

### PR DESCRIPTION
**Pull request by mattjoke**
* Session expiration dialog now does not show with 'invalid_token'
* fixed app undefined behaviour when the token expires
* Dialog now informs, that the user need to log in
* Updated czech translations

### CODE REVIEW

**Comment by mattjoke** [comment]
Probably, it would be wise to test it on the dev server.

File: apps/password-reset/src/assets/i18n/cs.json [hunk 1]
```diff
@@ -39,9 +39,9 @@
           "TITLE": "Požadovaný uživatel (dle ID nebo externí identity) neexistuje."
         },
         "SESSION_EXPIRATION": {
-          "TITLE": "Session expiration",
-          "DESCRIPTION": "Your session has expired. Please sign in to continue.",
-          "SIGN_IN": "Sign in"
+          "TITLE": "Platnost přihlášení vypršela",
+          "DESCRIPTION": "Byli jste automaticky odhlášen. Pro pokračování se znovu přihlašte.",
```
Comment (zlamalp) [inline apps/password-reset/src/assets/i18n/cs.json:43]: Either `Byl....odhlášen.` or `Byli....odhlášeni`. Also `přihlaste` and not `přihlašte`. Same for the user-profile app.

File: apps/admin-gui/src/app/app.component.ts [hunk 2]
```diff
@@ -63,7 +64,8 @@ export class AppComponent implements OnInit, AfterViewInit {
     private dialog: MatDialog,
     private router: Router,
     private initAuth: InitAuthService,
-    private cd: ChangeDetectorRef
+    private cd: ChangeDetectorRef,
+    private oatuth: OAuthService
```
Comment (bodnara) [inline apps/admin-gui/src/app/app.component.ts:68]: Just a typo.
```suggestion
    private oauth: OAuthService
```

File: apps/admin-gui/src/app/app.component.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
   {{ warningMessage }}
 </div>
 <perun-web-apps-notificator [displayWarning]="displayWarning"></perun-web-apps-notificator>
-<ng-template [ngIf]="!isLoginScreenShow && !isServiceAccess && !isServiceLogin()">
+<ng-template [ngIf]="!isLoginScreenShow && !isServiceAccess && !isServiceLogin() && !isLoggedIn()">
```
Comment (bodnara) [inline apps/admin-gui/src/app/app.component.html:6]: Please don't use function call in the html template, if some calculation needs to be made use a pipe.

File: libs/perun/utils/src/lib/perun-utils.ts [hunk 4]
```diff
@@ -864,3 +864,8 @@ export function isMemberIndirectString(member: RichMember): string {
   }
   return 'UNKNOWN';
 }
+
+export interface HttpErrorReponseError {
+  error: string;
+  error_description: string;
+}
```
Comment (bodnara) [inline libs/perun/utils/src/lib/perun-utils.ts:871]: I would say a better place for this would be in the perun-models lib.

**Comment by sarkapalkovicova** [comment]
I'm not sure why, but after login, the sidebar and icons from the header are missing.

image

**Comment by sarkapalkovicova** [comment]
I'm not sure why, but after login, the sidebar and icons from the header are missing.

image

File: apps/admin-gui/src/app/app.component.html (hunk 3 above)
Comment (mattjoke) [reply apps/admin-gui/src/app/app.component.html]: I have changed this to a pipe call. However, this change would not allow removing the sidebar and information about logged-in user (when revoking the tokens).

**Review by bodnara — APPROVED** [review]

**Review by Kuliak — APPROVED** [review]

File: libs/perun/services/src/lib/ApiInterceptor.ts [hunk 5]
```diff
@@ -178,6 +191,24 @@ export class ApiInterceptor implements HttpInterceptor {
 
   private handleInvalidAccessTokenError(err: HttpErrorResponse): void {
     if (err.status === 401) {
+      // We skip unnecessary login dialog
+      console.log(
+        this.initAuthService.isServiceAccessLoginScreenShown(),
+        this.initAuthService.isServiceAccess(),
+        this.isLoggedIn()
+      );
```
Comment (HejdaJakub) [inline libs/perun/services/src/lib/ApiInterceptor.ts:199]: Remove this testing console log, please.
```suggestion
```

File: libs/perun/models/src/index.ts [hunk 6]
```diff
@@ -16,3 +16,4 @@ export * from './lib/DynamicDataSource';
 export * from './lib/UserWithConsentStatus';
 export * from './lib/AuthPrivilege';
 export * from './lib/AppType';
+export * from './lib/HttpReponseError';
```
Comment (HejdaJakub) [inline libs/perun/models/src/index.ts:19]:
```suggestion
export * from './lib/HttpResponseError';
```

File: libs/perun/models/src/lib/HttpReponseError.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export interface HttpReponseError {
```
Comment (HejdaJakub) [inline libs/perun/models/src/lib/HttpReponseError.ts:1]:
```suggestion
export interface HttpResponseError {
```

File: libs/perun/services/src/lib/ApiInterceptor.ts [hunk 8]
```diff
@@ -9,7 +9,7 @@ import {
 } from '@angular/common/http';
 import { Observable, throwError } from 'rxjs';
 import { finalize, tap, switchMap, catchError } from 'rxjs/operators';
-import { RPCError } from '@perun-web-apps/perun/models';
+import { HttpReponseError, RPCError } from '@perun-web-apps/perun/models';
```
Comment (HejdaJakub) [inline libs/perun/services/src/lib/ApiInterceptor.ts:12]:
```suggestion
import { HttpResponseError, RPCError } from '@perun-web-apps/perun/models';
```

File: libs/perun/services/src/lib/ApiInterceptor.ts [hunk 9]
```diff
@@ -178,6 +191,24 @@ export class ApiInterceptor implements HttpInterceptor {
...
+      console.log(
+        this.initAuthService.isServiceAccessLoginScreenShown(),
+        this.initAuthService.isServiceAccess(),
+        this.isLoggedIn()
+      );
+      if (!this.isLoggedIn()) {
+        return;
+      }
+      // Skip 'invalid_token'
+      if ((err.error as HttpReponseError).error === 'invalid_token') {
```
Comment (HejdaJakub) [inline libs/perun/services/src/lib/ApiInterceptor.ts:204]:
```suggestion
      if ((err.error as HttpResponseError).error === 'invalid_token') {
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
Some small changes are requested, but the behavior is good now I think.

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**